pkg/evpn/bridge.go

```diff
@@ -0,0 +1,225 @@
+	// TODO: take Port from proto instead of hard-coded
+	if err := netlink.LinkAdd(vxlan); err != nil {
+		fmt.Printf("Failed to create Vxlan link: %v", err)
+		return nil, err
+	}
+	// Example: ip link set vxlan-<LB-vlan-id> master br-tenant addrgenmode none
+	if err := netlink.LinkSetMaster(vxlan, bridge); err != nil {
+		fmt.Printf("Failed to add Vxlan to bridge: %v", err)
+		return nil, err
+	}
+	// Example: ip link set vxlan-<LB-vlan-id> up
+	if err := netlink.LinkSetUp(vxlan); err != nil {
+		fmt.Printf("Failed to up Vxlan link: %v", err)
+		return nil, err
+	}
+	// Example: bridge vlan add dev vxlan-<LB-vlan-id> vid <LB-vlan-id> pvid untagged
+	if err := netlink.BridgeVlanAdd(vxlan, uint16(in.LogicalBridge.Spec.VlanId), true, true, false, false); err != nil {
+		fmt.Printf("Failed to add vlan to bridge: %v", err)
+		return nil, err
+	}
+	// TODO: bridge link set dev vxlan-<LB-vlan-id> neigh_suppress on
+	response := proto.Clone(in.LogicalBridge).(*pb.LogicalBridge)
+	response.Status = &pb.LogicalBridgeStatus{OperStatus: pb.LBOperStatus_LB_OPER_STATUS_UP}
+	s.Bridges[in.LogicalBridge.Name] = response
+	log.Printf("CreateLogicalBridge: Sending to client: %v", response)
+	return response, nil
+}
+
+// DeleteLogicalBridge deletes a LogicalBridge
+func (s *Server) DeleteLogicalBridge(_ context.Context, in *pb.DeleteLogicalBridgeRequest) (*emptypb.Empty, error) {
+	log.Printf("DeleteLogicalBridge: Received from client: %v", in)
+	// check required fields
+	if err := fieldbehavior.ValidateRequiredFields(in); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// Validate that a resource name conforms to the restrictions outlined in AIP-122.
+	if err := resourcename.Validate(in.Name); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// fetch object from the database
+	obj, ok := s.Bridges[in.Name]
+	if !ok {
+		if in.AllowMissing {
+			return &emptypb.Empty{}, nil
+		}
+		err := status.Errorf(codes.NotFound, "unable to find key %s", in.Name)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	resourceID := path.Base(obj.Name)
+	// use netlink to find vxlan device
+	vxlan, err := netlink.LinkByName(resourceID)
+	if err != nil {
+		err := status.Errorf(codes.NotFound, "unable to find key %s", resourceID)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// bring link down
+	if err := netlink.LinkSetDown(vxlan); err != nil {
+		fmt.Printf("Failed to up link: %v", err)
+		return nil, err
+	}
+	// delete bridge vlan
+	if err := netlink.BridgeVlanDel(vxlan, uint16(obj.Spec.VlanId), true, true, false, false); err != nil {
+		fmt.Printf("Failed to delete vlan to bridge: %v", err)
+		return nil, err
+	}
+	// use netlink to delete vxlan device
+	if err := netlink.LinkDel(vxlan); err != nil {
+		fmt.Printf("Failed to delete link: %v", err)
+		return nil, err
+	}
```

Comment:
Same comment as above regarding the presence of LogicalBridge.Spec.VNI field

Reply:
opened #86